spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/support/JdbcQueryMethod.java

```diff
@@ -53,13 +58,38 @@ public JdbcQueryMethod(Method method, RepositoryMetadata metadata, ProjectionFac
 	 *
 	 * @return May be {@code null}.
 	 */
+
 	@Nullable
 	String getAnnotatedQuery() {
-		return getMergedAnnotationAttribute("value");
+		String annotatedValue = getQueryValue();
+		return StringUtils.hasText(annotatedValue) ? annotatedValue : getValueNamedQuery();
 	}
 
 	/**
-	 * Returns the class to be used as {@link org.springframework.jdbc.core.RowMapper}
+	 * Returns the annotated query with key value if it exists.
+	 *
+	 * @return May be {@code null}.
+	 */
+	@Nullable
+	public String getQueryValue() {
+		return getMergedAnnotationAttribute("value");
+	}
+	/**
+	 * Returns the annotated query with key name if it exists.
+	 *
+	 * @return May be {@code null}.
+	 */
+	@Nullable
+	public String getValueNamedQuery() {
+		String annotatedName = getMergedAnnotationAttribute("name");
```

Comment:
Sorry, it seems I wasn't clear here. 

If no name is specified, the method name should be used a the name for the named query.